### PR TITLE
Change __init to init for objects and modules

### LINKED ISSUE #23358: Change __init to init for objects and modules
Refer Commit https://github.com/ballerina-platform/ballerina-spec/commit/86e30bb40eefd6b54a7281b96be59375f43b9408

### PR DESCRIPTION
**Pull request by chiranSachintha**
Fixes #23358 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/http/src/main/ballerina/src/http/non_listening_service_endpoint.bal [hunk 1]
```diff
@@ -44,13 +44,9 @@ public type MockListener object {
...
     }
 
-    public function __init(int port, ListenerConfiguration? config = ()) {
+    public function init(int port, ListenerConfiguration? config = ()) {
         self.config = config ?: {};
         self.port = port;
-        self.init(self.config);
-    }
-
-    public function init(ListenerConfiguration c) {
```
Comment (KRVPerera) [inline stdlib/http/src/main/ballerina/src/http/non_listening_service_endpoint.bal:53]: Looks like we are losing a public function here.

File: stdlib/http/src/main/ballerina/src/http/service_endpoint.bal [hunk 2]
```diff
@@ -78,18 +78,12 @@ public type Listener object {
...
         self.instanceId = system:uuid();
         self.config = config ?: {};
         self.port = port;
-        self.init(self.config);
-    }
-
-    # Gets invoked during module initialization to initialize the endpoint.
-    #
-    # + c - Configurations for HTTP service endpoints
-    public function init(ListenerConfiguration c) {
```
Comment (KRVPerera) [inline stdlib/http/src/main/ballerina/src/http/service_endpoint.bal:91]: Looks like we are losing a public function here.

File: stdlib/http/src/main/ballerina/src/http/non_listening_service_endpoint.bal (hunk 1 above)
Comment (chiranSachintha) [reply stdlib/http/src/main/ballerina/src/http/non_listening_service_endpoint.bal:53]: I asked from stdlib-team and merged init and __init functions.

**Comment by MaryamZi** [comment]
@nadeeshaan FYI please. This PR introduces some langserver changes for the change from `__init()` to `init()` for the module/object init.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java [hunk 3]
```diff
@@ -288,8 +288,8 @@ private static void createTypesInstance(ClassWriter cw, List<BIRTypeDefinition>
                     addObjectFields(mv, objectType.fields);
                     BObjectTypeSymbol objectTypeSymbol = (BObjectTypeSymbol) objectType.tsymbol;
                     addObjectInitFunction(mv, objectTypeSymbol.generatedInitializerFunc, objectType, indexMap,
-                            "$__init$", "setGeneratedInitializer", symbolTable);
-                    addObjectInitFunction(mv, objectTypeSymbol.initializerFunc, objectType, indexMap, "__init",
+                            "$init$", "setGeneratedInitializer", symbolTable);
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java:291]: @Kishanthan, please note this PR will be removing the `__` part from the init functions.

File: stdlib/websub/src/main/ballerina/src/websub/subscriber_service_endpoint.bal [hunk 4]
```diff
@@ -39,8 +39,22 @@ public type Listener object {
     #
     # + port - The port number of the remote service
     # + config - The configurations related to the `websub:Listener`
-    public function __init(int port, SubscriberListenerConfiguration? config = ()) {
-        self.init(port, config);
+    # 
+    # Gets called when the `websub:Listener` is being initialized during the module initialization.
```
Comment (chiranSachintha) [inline stdlib/websub/src/main/ballerina/src/websub/subscriber_service_endpoint.bal:43]: @MaryamZi Can we also remove these documentation lines?

Comment (MaryamZi) [reply stdlib/websub/src/main/ballerina/src/websub/subscriber_service_endpoint.bal:43]: Yes this too.

Docs take the following format.

```
    # Description.
    #
    # + p1 - Description for param 1.
    # + p2 - Description for param 2.
    public function init(int param1, string? param2 = ()) {
    }
```

After the params (or fields in a record/object type) you'll only have deprecation documentation.

File: stdlib/http/src/main/ballerina/src/http/service_endpoint.bal [hunk 5]
```diff
@@ -77,19 +77,10 @@ public type Listener object {
     # Gets invoked during module initialization to initialize the listener.
     #
     # + port - Listening port of the HTTP service listener
-    # + c - Configurations for HTTP service listener
-    public function __init(int port, public ListenerConfiguration? config = ()) {
```
Comment (MaryamZi) [inline stdlib/http/src/main/ballerina/src/http/service_endpoint.bal:81]: We need the docs for config here right?

Let's add
```
# + config - Configurations for HTTP service listener
```

Comment (chiranSachintha) [reply stdlib/http/src/main/ballerina/src/http/service_endpoint.bal:81]: Done

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java (hunk 3 above)
Comment (Kishanthan) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java:293]: ack, as long as it does not break anything :). pl check for other places in the codegen as well.

Comment (chiranSachintha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java:293]: @Kishanthan All other places were updated.

**Pull request merged**